src/core/matchers/matchersUtil.js

```diff
@@ -167,11 +167,19 @@ getJasmineRequireObj().matchersUtil = function(j$) {
 
     if (result) {
       // Objects with different constructors are not equivalent, but `Object`s
-      // from different frames are.
-      var aCtor = a.constructor, bCtor = b.constructor;
-      if (aCtor !== bCtor && !(isFunction(aCtor) && (aCtor instanceof aCtor) &&
-        isFunction(bCtor) && (bCtor instanceof bCtor))) {
-        return false;
+      // or `Array`s from different frames are.
+      var areArrays = className === '[object Array]';
+      if (!areArrays) {
+        if (typeof a != 'object' || typeof b != 'object') {
+          return false;
+        }
+
+        var aCtor = a.constructor, bCtor = b.constructor;
+        if (aCtor !== bCtor && !(isFunction(aCtor) && aCtor instanceof aCtor &&
+               isFunction(bCtor) && bCtor instanceof bCtor) &&
+               ('constructor' in a && 'constructor' in b)) {
```

Comment:
Not sure, it was present in the underscore `isEqual` function (https://github.com/jashkenas/underscore/blob/95337a876784d5df7e62b2e55080ddf9199b6e9b/underscore.js#L1127). Tests pass without it, so I'll removed it.